test/fixtures/dialects/tsql/inner_joins.sql

```diff
@@ -0,0 +1,7 @@
+SELECT
+    1 AS RegionCode
+FROM BA 
+LEFT OUTER JOIN I 
```

Comment:
I would consider these joins to be chained one after another, not nested. It's kinda similar to arithmetic, e.g. `1 + 2 + 3 + 4`. If I parse `SELECT 1 + 2 + 3 + 4`, it looks like:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    expression:
[L:  1, P:  8]      |                        literal:                              '1'
[L:  1, P:  9]      |                        whitespace:                           ' '
[L:  1, P: 10]      |                        binary_operator:                      '+'
[L:  1, P: 11]      |                        whitespace:                           ' '
[L:  1, P: 12]      |                        literal:                              '2'
[L:  1, P: 13]      |                        whitespace:                           ' '
[L:  1, P: 14]      |                        binary_operator:                      '+'
[L:  1, P: 15]      |                        whitespace:                           ' '
[L:  1, P: 16]      |                        literal:                              '3'
[L:  1, P: 17]      |                        whitespace:                           ' '
[L:  1, P: 18]      |                        binary_operator:                      '+'
[L:  1, P: 19]      |                        whitespace:                           ' '
[L:  1, P: 20]      |                        literal:                              '4'
[L:  1, P: 21]      |            [META] dedent:
[L:  1, P: 21]      |    newline:                                                  '\n'
```

Could we organize it to parse that way instead? That seems more correct.

Reply:
Oh, never mind. It _is_ nested, because of the sequence of the `ON` clauses. Tricky!!

Reply:
Yup exactly. I was confused at first why it didn’t just work until I saw that.